docs/concepts/tokenizables.rst

```diff
@@ -0,0 +1,374 @@
+
+Deduplication of ``GufeTokenizable``\s
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are two types of deduplication of ``GufeTokenizable``\s:
+* Objects are deduplicated in memory because **gufe** keeps a registry of all instantiated ``GufeTokenizable``\s.
+* Objects can be deduplicated on storage to disk because we store by reference to the **gufe** key.
+
+.. _gufe-memory-deduplication:
+
+Deduplication in memory (flyweight pattern)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Memory deduplication means that only one object with a given ``GufeKey`` will exist in any single Python session.
+We ensure this by maintaining a registry of all ``GufeTokenizable``\s that gets updated any time a ``GufeTokenizable`` is created.
+The registry is a mapping to weak references, which allows Python's garbage collection to clean up ``GufeTokenizable``\s that are no longer needed.
+This is essentially an implementation of the `flyweight pattern <https://en.wikipedia.org/wiki/Flyweight_pattern>`_.
+
+This memory deduplication is ensured by the ``GufeTokenizable.from_dict``, which is typically used in deserialization.
+It will always use the first object in memory with that ``GufeKey``.
+This can lead to some unexpected behavior; for example, using the ``Foo`` class defined above:
+
+.. code-block:: python
+
+    # here Foo is a GufeTokenizable:
+    >>> a = Foo(0)
+    >>> b = Foo(0)
+    >>> a is b
+    False
+    >>> c = Foo.from_dict(a.to_dict())
+    >>> c is a  # surprise!
+    True
+    >>> d = Foo.from_dict(b.to_dict())
+    >>> d is b
+    False
+    >>> d is a  # this is because `a` has the spot in the registry
+    True
+
+
+Deduplication on disk
+~~~~~~~~~~~~~~~~~~~~~
+
+Deduplication on disk storage is fundamentally the responsibility of the specific storage system, which falls outside the scope of **gufe**.
+However, **gufe** provides some tools to facilitate implementation of a storage system.
+
+The main idea is to use the ``GufeKey`` to ensure uniqueness, and to use it as a label for the object's serialized representation.
+Additionally, the ``GufeKey``, which is simply a string, can be used as a stand-in for the object.
+When an outer ``GufeTokenizable`` contains an inner ``GufeTokenizable``, the outer can store the ``GufeKey`` in place of the inner object.
+That is, we can store by reference to the ``GufeKey``.
+
+To convert a ``GufeTokenizable`` ``obj`` into a dictionary that references inner ``GufeTokenizable``\s by ``GufeKey``, use ``obj.to_keyed_dict()``.
+That method replaces each ``GufeTokenizable`` by a ``dict`` with a single key, ``':gufe-key:'``, mapping to the ``GufeKey`` of the object.
+Of course, you'll also need to do the same for all inner ``GufeTokenizables``; to get a list of all of them, use :func:`.get_all_gufe_objs` on the outermost ``obj``.
+
+.. TODO: add a tutorial for this in the tutorials section?
+
+
+.. _serialization:
+
+1. Serializable Representations of ``GufeTokenizable``\s
```

Comment:
```suggestion
3. Serializable Representations of ``GufeTokenizable``\s
```

Reply:
I'll try to find how to turn off this auto-numbering (if it's possible)